src/controllers/profileController.ts

```diff
@@ -104,7 +115,57 @@ export default class ProfileController {
     | IResponseErrorTooManyRequests
     | IResponseSuccessAccepted
   > =>
-    withUserFromRequest(req, async user =>
-      this.profileService.emailValidationProcess(user)
-    );
+    withUserFromRequest(req, async user => {
+      type ErrorTypes =
+        | IResponseErrorValidation
+        | IResponseErrorNotFound
+        | IResponseErrorInternal
+        | IResponseErrorTooManyRequests;
+      const startEmailValidationProcessTask = tryCatch(
+        () => this.profileService.emailValidationProcess(user),
+        () => new Error("Error starting email validation process")
+      ).foldTaskEither<ErrorTypes, IResponseSuccessAccepted>(
+        _ => fromLeft(ResponseErrorInternal(_.message)),
+        response =>
+          response.kind !== "IResponseSuccessAccepted"
+            ? fromLeft(response)
+            : taskEither.of(response)
+      );
+
+      return tryCatch(
+        () =>
+          // check if an email validation process exists in Redis
+          this.sessionStorage.isEmailValidationProcessPending(user.fiscal_code),
+        toError
+      )
+        .chain(fromEither)
+        .foldTaskEither(
+          () => startEmailValidationProcessTask,
+          isEmailValidationProcessPending =>
+            isEmailValidationProcessPending
+              ? // email validation process already requested, return 202
```

Comment:
I guess we have to change the api specs as well

Reply:
What do you mean? `startEmailValidation` already returns 202 in case of success :)

Reply:
ok!